src/emc/rs274ngc/interp_internal.cc

```diff
@@ -476,14 +476,14 @@ int Interp::call_level(void) { return _setup.call_level; }
 
 std::string toString(GCodes g)
 {
-    char buf[10]={};
+    char buf[15]={};
     int dec_value = g%10;
     if (dec_value)
     {
         // Has a decimal
-        snprintf(buf, 10, "G%d.%d", g/10, dec_value);
```

Comment:
instead of a constant, better to write sizeof(buf).

Since this is C++ code returning a std::string, you could consider using boost::format here (though we don't use it yet).  `return (boost::format("G%d.%d") % (g/10) % (dec_value)).str()` or so.